### DISCUSSION TITLE
Feature Requests: 2021-08 -- 2021-12

### DISCUSSION THREAD

**Discussion by stephenroller** [facebookresearch/ParlAI#3933] [Ideas]
If you have requests for support of certain things in ParlAI, please put them here. We do not make promises about completion, but are eager to use this to gauge what the community would find most helpful.

**Comment by jh-skaion** [comment]
@stephenroller is there plan to integrate parlai's chatbot with knowledge graph db?

**Reply by Adrian-1234** [reply]
Me too.

**Reply by jh-skaion** [reply]
@stephenroller any plan on this? i have read multiple paper regarding using knowledge base\knowledge graph in dialog system and it seems a very interesting idea

**Reply by Adrian-1234** [reply]
So following on from my 'Me too' above, I decided to get coding on this problem.   I have modified parts of the Parlai  code to interact with my private General Knowledge Graph DB. $search_query is sent to the Graph and results are returned.   These results are prepended to the internet query results. This then works really well when the Graph nodes contain good correlations to the $search_query. I also return the graph path (n) -[r]-(m) which gets translated into an English result for broader matches when again prepended to the results.
The problem is when there is little correlation with existing nodes.   I suspect that the solution to this might be to use NER's from the input string for matches within the Graph,  and then search for multi-node and edge correlations within the Graph.   This is a much more challenging problem. I am starting to experiment with the latter to leverage the Graph further.
I'm not planning to release the code as I'm guessing that Facebook is sensitive towards this type of development given that Blender 2.0 was released as an experiment, but it is interesting to explore the art of the possible!

**Reply by stephenroller** [reply]
I suspect that does work pretty well. If there are existing public datasets that meet this need, I think we could add them. Otherwise, I don't think we will have the bandwidth to conduct novel research for this.

**Comment by robbine** [comment]
is there plan to integrate parlai's chatbot with task oriented dialog systems like rasa ?

**Reply by stephenroller** [reply]
We will be adding support for e2e task oriented dialogue.

**Reply by stephenroller** [reply]
Update: #4082 and #4083.
